sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceResponse.java

```diff
@@ -135,18 +138,12 @@ public String getResponseBodyAsString() {
                 // Aggregate on single partition collection may return the aggregated value only
                 // In that case it needs to encapsulated in a special document
 
-                String resourceJson = jToken.isValueNode() || jToken.isArray()// to add nulls, arrays, objects
-                        ? String.format("{\"%s\": %s}", Constants.Properties.VALUE, jToken.toString())
-                                : toJson(jToken);
-                        T resource = null;
-                        try {
-                            resource = c.getConstructor(String.class).newInstance(resourceJson);
-                        } catch (InstantiationException | IllegalAccessException | IllegalArgumentException
-                                | InvocationTargetException | NoSuchMethodException | SecurityException e) {
-                            throw new IllegalStateException("Failed to instantiate class object.", e);
-                        }
-
-                        queryResults.add(resource);
+                JsonNode resourceJson = jToken.isValueNode() || jToken.isArray()// to add nulls, arrays, objects
+                        ? fromJson(String.format("{\"%s\": %s}", Constants.Properties.VALUE, jToken.toString()))
+                                : jToken;
+
+               T resource = (T) BridgeInternal.instantiateJsonSerializable((ObjectNode) resourceJson, c);
```

Comment:
When we down cast JsonNode to ObjectNode we should make sure it is never an ArrayNode. Why don't we just use it as JsonNode itself and why do we need to convert it to ObjectNode any reason?